### PR TITLE
build: make shared object version fetching generic

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: bindings/python/tools/python/setup.cfg.in [hunk 1]
```diff
@@ -26,4 +26,4 @@ include_package_data = True
 install_requires = file: requirements.txt
 
 [options.package_data]
-ostk.io=libopen-space-toolkit-io.so.0, ${LIBRARY_TARGET}.*${EXTENSION}*.so
+ostk.io=libopen-space-toolkit-io${SHARED_LIBRARY_TARGET}.${PROJECT_VERSION_MAJOR}, ${LIBRARY_TARGET}.*${EXTENSION}*.so
```
Comment (Derollez) [inline bindings/python/tools/python/setup.cfg.in:29]:
```suggestion
ostk.io=${SHARED_LIBRARY_TARGET}.${PROJECT_VERSION_MAJOR}, ${LIBRARY_TARGET}.*${EXTENSION}*.so
```

File: tools/development/start.sh [hunk 2]
```diff
@@ -55,7 +55,7 @@ if [[ ! -z ${1} ]] && [[ ${1} == "--link" ]]; then
             cp -as /mnt/open-space-toolkit-core/include/OpenSpaceToolkit/Core /usr/local/include/OpenSpaceToolkit/Core; \
             cp -as /mnt/open-space-toolkit-core/src/OpenSpaceToolkit/Core/* /usr/local/include/OpenSpaceToolkit/Core/; \
             ln -s /mnt/open-space-toolkit-core/lib/libopen-space-toolkit-core.so /usr/local/lib/; \
-            ln -s /mnt/open-space-toolkit-core/lib/libopen-space-toolkit-core.so.0 /usr/local/lib/; \
+            ln -s /mnt/open-space-toolkit-core/lib/libopen-space-toolkit-core${SHARED_LIBRARY_TARGET}.${PROJECT_VERSION_MAJOR} /usr/local/lib/; \
```
Comment (Derollez) [inline tools/development/start.sh:58]:
```suggestion
            ln -s /mnt/open-space-toolkit-core/lib/${SHARED_LIBRARY_TARGET}.${PROJECT_VERSION_MAJOR} /usr/local/lib/; \
```
Also I am not sure those env vars can be obtained in here 🤔

Comment (vishwa2710) [reply tools/development/start.sh:58]: Yeah they might have to be passed... I'll check if it works

Comment (vishwa2710) [reply tools/development/start.sh:58]: Okay fixed it, thank you for the thorough review on this.

**Pull request merged**